language/move-core/types/src/account_address.rs

```diff
@@ -45,6 +45,11 @@ impl AccountAddress {
         )
     }
 
+    pub fn short_str_lossless(&self) -> String {
```

Comment:
It would probably be good to add some Rust unit tests for this (putting `#[test]`'s at the bottom of the file will work just fine for this).